tests/test_all_rules.py

```diff
@@ -1215,28 +1215,27 @@ def test_if_plugins_explicitly_defined(self):
 class TestAlertSuppression(BaseRuleTest):
     """Test rule alert suppression."""
 
-    @unittest.skipIf(PACKAGE_STACK_VERSION < Version.parse("8.8.0"),
-                     "Test only applicable to 8.6+ stacks for rule alert suppression feature.")
-    def test_group_length(self):
-        """Test to ensure the rule alert suppression group_by does not exceed 3 elements."""
-        for rule in self.production_rules:
-            if rule.contents.data.get('alert_suppression'):
-                group_length = len(rule.contents.data.alert_suppression.group_by)
-                if group_length > 3:
-                    self.fail(f'{self.rule_str(rule)} has rule alert suppression with more than 3 elements.')
-
     @unittest.skipIf(PACKAGE_STACK_VERSION < Version.parse("8.8.0"),
                      "Test only applicable to 8.8+ stacks for rule alert suppression feature.")
     def test_group_field_in_schemas(self):
         """Test to ensure the fields are defined is in ECS/Beats/Integrations schema."""
         for rule in self.production_rules:
-            if rule.contents.data.get('alert_suppression'):
-                group_by_fields = rule.contents.data.alert_suppression.group_by
+            rule_type = rule.contents.data.get('type')
+            if rule_type in ('query', 'threshold') and rule.contents.data.get('alert_suppression'):
+                if isinstance(rule.contents.data.alert_suppression, AlertSuppressionMapping):
+                    group_by_fields = rule.contents.data.alert_suppression.group_by
+                elif isinstance(rule.contents.data.alert_suppression, ThresholdAlertSuppression):
+                    group_by_fields = rule.contents.data.threshold.field
                 min_stack_version = rule.contents.metadata.get("min_stack_version")
                 if min_stack_version is None:
                     min_stack_version = Version.parse(load_current_package_version(), optional_minor_and_patch=True)
                 else:
                     min_stack_version = Version.parse(min_stack_version)
+
+                if min_stack_version < Version.parse("8.12.0") and rule_type == 'threshold':
```

Comment:
you can probably remove this since it should never pass validation

Reply:
I thought that was the case too with the min_compat but those mixins didn't appear to validate against the rules minstack, only the latest version in our package.

Reply:
I rechecked the unit tests and `test_rule_backports_for_restricted_fields` should cover this. So you're right. We're good to remove this.